src/kernels/jupyter/launcher/notebookStarter.node.ts

```diff
@@ -198,11 +198,6 @@ export class NotebookStarter implements Disposable {
         const promisedArgs: Promise<string>[] = [];
         promisedArgs.push(Promise.resolve('--no-browser'));
         promisedArgs.push(Promise.resolve(this.getNotebookDirArgument(workingDirectory)));
-        // When kernels fail to start, Jupyter will attempt to restart 5 times,
-        // & this is very slow (we dont want users to wait because of kernel failures).
-        // Also when kernels die, we don't restart automatically with raw kernels,
-        // We should'nt do the same with jupyter (else startup code will not run).
-        promisedArgs.push(Promise.resolve('--KernelManager.autorestart=False'));
```

Comment:
Reverting this change, else this breaks jupyter.
I.e. if a kernel dies, then we never get a response back for anything, we don't know whether the died or not.
With auto start enabled, jupyter will at a minimum re-start the kernel and we know that it was auto-restarted as the kernel died. Note: Jupyter auto restarts when kernels die.

Hence this change is required (& it was wrong to add)